music_assistant/server/providers/deezer/__init__.py

```diff
@@ -554,64 +596,80 @@ def parse_track(
-
-    async def search_and_parse_artists(self, query: str, limit: int = 5) -> list[Artist]:
+        deezer_tracks = await self.client.search(query=query, limit=limit)
+        tracks = []
+        index = 0
+        async for track in deezer_tracks:
+            tracks.append(self.parse_track(track, user_country))
+            index += 1
+            if index >= limit:
+                return tracks
+        return tracks
+
+    async def search_and_parse_artists(self, query: str, limit: int = 20) -> list[Artist]:
         """Search for artists and parse them."""
-        deezer_artist = await self.client.search_artist(query=query, limit=limit)
-        return [self.parse_artist(artist=artist) for artist in deezer_artist]
-
-    async def search_and_parse_albums(self, query: str, limit: int = 5) -> list[Album]:
+        deezer_artist = await self.client.search_artists(query=query, limit=limit)
+        artists = []
+        index = 0
+        async for artist in deezer_artist:
+            artists.append(self.parse_artist(artist))
+            index += 1
+            if index >= limit:
+                return artists
+        return artists
+
+    async def search_and_parse_albums(self, query: str, limit: int = 20) -> list[Album]:
         """Search for album and parse them."""
-        deezer_albums = await self.client.search_album(query=query, limit=limit)
-        return [self.parse_album(album=album) for album in deezer_albums]
+        deezer_albums = await self.client.search_albums(query=query, limit=limit)
+        albums = []
+        index = 0
+        async for album in deezer_albums:
+            albums.append(self.parse_album(album))
+            index += 1
+            if index >= limit:
+                return albums
+        return albums
 
-    async def search_and_parse_playlists(self, query: str, limit: int = 5) -> list[Playlist]:
+    async def search_and_parse_playlists(self, query: str, limit: int = 20) -> list[Playlist]:
         """Search for playlists and parse them."""
-        deezer_playlists = await self.client.search_playlist(query=query, limit=limit)
-        return [self.parse_playlist(playlist=playlist) for playlist in deezer_playlists]
-
-    ### OTHER PARSING FUNCTIONS ###
-    def _get_album(self, track: deezer.Track) -> Album | None:
-        try:
-            return self.parse_album(album=track.get_album())
-        except AttributeError:
-            return None
+        deezer_playlists = await self.client.search_playlists(query=query, limit=limit)
+        playlists = []
+        index = 0
+        async for playlist in deezer_playlists:
+            playlists.append(self.parse_playlist(playlist))
+            index += 1
+            if index >= limit:
+                return playlists
+        return playlists
 
     ### OTHER FUNCTIONS ###
     async def update_access_token(self, app_id, app_secret, code, http_session=None) -> str:
```

Comment:
Why are `update_access_token` and `_post_http` part of the class? `_post_http` is used in the `update_access_token` function only and that function is used from outside the class only as far as I can see. For that reason it makes much more sense to move them out of the class. This way you can also use them more easily in the `get_config_entries` function 🙂

Reply:
I think i was moving things around and didnt think about it. But good point im moving it now